set.mm

```diff
@@ -465681,6 +465847,107 @@ its graph has a given second element (that is, function value).
+
+  ${
+    $d A x $.  $d F x $.  $d V x $.  $d W x $.  $d Z x $.
+    fdifsuppconst.1 $e |- A = ( dom F \ ( F supp Z ) ) $.
+    $( A function is a zero constant outside of its support.  (Contributed by
+       Thierry Arnoux, 22-Jun-2024.) $)
+    fdifsuppconst $p |- ( ( Fun F /\ F e. V /\ Z e. W )
+                       -> ( F |` A ) = ( A X. { Z } ) ) $=
+      ( vx wfun wcel cres csn cxp wceq wa cdm wfn biimpi adantl cfv cvv co cdif
+      funfn ad2antrr csupp difssd eqsstrid fnssresd fnconstg cv adantr ad3antlr
+      dmexg simplr eleq2i fvdifsupp simpr fvresd adantll 3eqtr4d eqfnfvd 3impa
+      fvconst2g ) BHZBCIZEDIZBAJZAEKLZMVDVENZVFNZGAVGVHVJBOZABVDBVKPZVEVFVDVLBU
+      CQUDZVJAVKBEUEUAZUBZVKFVJVKVNUFUGUHVFVHAPVIAEDUIRVJGUJZAIZNZVPBSEVPVGSVPV
+      HSZVRVKBTDVPEVJVLVQVMUKVEVKTIVDVFVQBCUMULVIVFVQUNVQVPVOIZVJVQVTAVOVPFUOQR
+      UPVRVPABVJVQUQURVFVQVSEMVIAEVPDVCUSUTVAVB $.
+  $}
+
+  ${
+    $d .0. x y $.  $d F x y $.  $d V x y $.  $d W x y $.
+    $( The range of a function restricted to its support.  (Contributed by
+       Thierry Arnoux, 25-Jun-2024.) $)
+    ressupprn $p |- ( ( Fun F /\ F e. V /\ .0. e. W )
+                   -> ran ( F |` ( F supp .0. ) ) = ( ran F \ { .0. } ) ) $=
+      ( vy vx wcel crn cv cfv wceq wrex wne wa wfn cvv anbi1d pm5.32da fvelrnb
+      wb wfun w3a csupp co cres cdif funfn biimpi 3ad2ant1 dmexg 3ad2ant2 simp3
+      csn cdm elsuppfn syl3anc anass a1i biimprd impl fvresd eqeq1d ancom simpr
+      neeq1d syl5bb bitrd rexbidv2 wss suppssdm fnssres sylancl eldifsn r19.41v
+      3bitrd syl 3bitr4g 3bitr4d eqrdv ) AUAZABGZDCGZUBZEAADUCUDZUEZHZAHZDUMUFZ
+      WCFIZWEJZEIZKZFWDLZWIAJZWKKZWKDMZNZFAUNZLZWKWFGZWKWHGZWCWLWQFWDWRWCWIWDGZ
+      WLNWIWRGZWNDMZNZWLNZXCXDWLNZNZXCWQNWCXBXEWLWCAWROZWRPGZWBXBXETVTWAXIWBVTX
+      IAUGUHUIZWAVTXJWBABUJUKVTWAWBULWIAPCWRDUOUPZQXFXHTWCXCXDWLUQURWCXCXGWQWCX
+      CNZXGXDWONZWQXMXDWLWOXMXDNZWJWNWKXOWIWDAWCXCXDXBWCXBXEXLUSUTVAVBRXNWOXDNX
+      MWQXDWOVCXMWOXDWPXMWONWNWKDXMWOVDVERVFVGRVOVHWCWEWDOZWTWMTWCXIWDWRVIXPXKA
+      DVJWRWDAVKVLFWDWKWESVPWCXIXAWSTXKXIWKWGGZWPNWOFWRLZWPNXAWSXIXQXRWPFWRWKAS
+      QWKWGDVMWOWPFWRVNVQVPVRVS $.
+  $}
+
+  $( Express the support of a function as the preimage of its range except
+     zero.  (Contributed by Thierry Arnoux, 24-Jun-2024.) $)
+  supppreima $p |- ( ( Fun F /\ F e. V /\ Z e. W )
+                            -> ( F supp Z ) = ( `' F " ( ran F \ { Z } ) ) ) $=
+    ( wfun wcel w3a ccnv crn cima csn cdif cdm csupp co wceq cnvimarndm difeq1d
+    a1i difpreima 3ad2ant1 wss suppssdm dfss4 mpbi suppiniseg difeq2d 3eqtr4rd
+    syl5eqr ) AEZABFZDCFZGZAHZAIZJZUNDKZJZLZAMZURLZUNUOUQLJZADNOZUMUPUTURUPUTPU
+    MAQSRUJUKVBUSPULUOUQATUAUMVCUTUTVCLZLZVAVCUTUBVEVCPADUCVCUTUDUEUMVDURUTABCD
+    UFUGUIUH $.
+
+  $( Finite support implies finite range, when the zero is excluded.
+     (Contributed by Thierry Arnoux, 24-Jun-2024.) $)
+  fsupprnfi $p |- ( ( ( Fun F /\ F e. V ) /\ ( .0. e. W /\ F finSupp .0. ) )
```

Comment:
The range is still finite when the zero is not excluded, isn't it?

Reply:
Correct, that's a much better and simpler statement, thanks. I'll update the theorem.